tests/components/homematicip_cloud/test_climate.py

```diff
@@ -219,6 +224,19 @@ async def test_hmip_heating_group_heat(
     # Only fire event from last async_manipulate_test_data available.
     assert hmip_device.mock_calls[-1][0] == "fire_update_event"
 
+    await hass.services.async_call(
+        "climate",
+        "set_preset_mode",
+        {"entity_id": entity_id, "preset_mode": PRESET_ECO},
+        blocking=True,
+    )
+    assert len(hmip_device.mock_calls) == service_call_counter + 25
+    assert hmip_device.mock_calls[-1][0] == "set_control_mode"
+    assert hmip_device.mock_calls[-1][1] == ("ECO",)
+    await async_manipulate_test_data(hass, hmip_device, "controlMode", "ECO")
+    ha_state = hass.states.get(entity_id)
+    assert ha_state.attributes[ATTR_PRESET_MODE] == PRESET_ECO
+
```

Comment:
Can we add here which state the entity has? As in it looks in the code above as this preset is actually a mode for the device in real life?

Reply:
The state should be Auto. Should i add an assert statement to check that?

Reply:
Yes. Both before and after setting the preset would be ideal to see that the state doesn't change.